### DISCUSSION TITLE
Result comparison with EAS Build

### DISCUSSION THREAD

**Discussion by brentvatne** [cheeaun/hackerweb-native-2#5] [General]
Hi there!

I saw your post on Reddit and the concern about app size. I thought I'd use this as an example to try out the new [EAS Build](https://blog.expo.io/expo-application-services-eas-build-and-submit-fc1d1476aa2e) preview and the early support it has for managed workflow.

The app worked with EAS Build without any changes required (beyond creating `eas.json` when running `eas build`), and the install size was reduced to ~6.7mb on my iPhone 11 Pro, down from 52.6mb reported from the App Store for the expo build version of the app.

Here's a screenshot of the build info in App Store Connect, and a video of the result (mostly indistinguishable from the source, I just renamed the app and you'll notice in the settings there is no Expo SDK version present in the build).

![IMG_8242](https://user-images.githubusercontent.com/90494/102924236-06f43800-4446-11eb-96b4-a52247b5088e.jpg)

https://user-images.githubusercontent.com/90494/102924414-605c6700-4446-11eb-8abd-60c157979e4c.mp4

<hr />

Basically what EAS Build does for managed projects is "eject" the project, configure everything based on your app config and installed packages, then set up the credentials and run a build with `fastlane gym`.

This took about 11 minutes (after brief queuing):

<img width="1092" alt="Screen Shot 2020-12-22 at 11 25 02 AM" src="https://user-images.githubusercontent.com/90494/102925526-64898400-4448-11eb-8c24-8b0e7c5b3fea.png">

If you want free early access to this so you can try it out and give feedback, just DM me on Twitter @notbrent and let me know what your Expo username is and I can add you :)

(we'll have a free tier that should be more than enough for hobby/open source apps like this when we launch General Availability)

**Comment by cheeaun** [comment]
Nice!! DM-ed 😀

**Comment by cheeaun** [comment]
Just released 2.0.1 (Build 13) on TestFlight 👉 https://testflight.apple.com/join/PHcLooxC

Universal binary download size reduced by **19X** and install size reduced by **13X**, compared to last production 2.0.0 build.

Comparison tables below for 2.0.0 (Build 9, Expo 40), 2.0.1 (Build 11, Expo 41) and 2.0.1 (Build 13, Expo 41, EAS).

# 2.0.0 (Build 9) Production

> Expo 40, Managed workflow, non-EAS

| DEVICE TYPE                                            | DOWNLOAD SIZE | INSTALL SIZE |
| ------------------------------------------------------ | ------------- | ------------ |
| Universal                                              | 71440 KB      | 86227 KB     |
| iPad (4th generation)                                  | 39325 KB      | 48480 KB     |
| iPad (4th generation) Wi‑Fi + Cellular                 | 39325 KB      | 48480 KB     |
| iPad mini 4                                            | 41773 KB      | 53053 KB     |
| iPad mini 4 Wi‑Fi + Cellular                           | 41773 KB      | 53053 KB     |
| iPad Air 2                                             | 41773 KB      | 53053 KB     |
| iPad Air 2 Wi‑Fi + Cellular                            | 41773 KB      | 53053 KB     |
| iPad (5th generation)                                  | 41773 KB      | 53053 KB     |
| iPad (5th generation) Wi‑Fi + Cellular                 | 41773 KB      | 53053 KB     |
| iPad Pro (12.9‑inch) (2nd generation)                  | 41772 KB      | 53053 KB     |
| iPad Pro (12.9‑inch) (2nd generation) Wi‑Fi + Cellular | 41772 KB      | 53053 KB     |
| iPad Pro (10.5‑inch)                                   | 41772 KB      | 53053 KB     |
| iPad Pro (10.5‑inch) Wi‑Fi + Cellular                  | 41772 KB      | 53053 KB     |
| iPad (6th generation)                                  | 41773 KB      | 53053 KB     |
| iPad (6th generation) Wi‑Fi + Cellular                 | 41773 KB      | 53053 KB     |
| iPad Pro (11‑inch) (2nd generation) Wi‑Fi + Cellular   | 22975 KB      | 52595 KB     |
| iPad Pro (12.9‑inch) (4th generation)                  | 22975 KB      | 52595 KB     |
| iPad Pro (12.9‑inch) (4th generation) Wi‑Fi + Cellular | 22975 KB      | 52595 KB     |
| iPhone 5                                               | 39318 KB      | 48480 KB     |
| iPhone 5c                                              | 39318 KB      | 48480 KB     |
| iPhone 5S                                              | 41773 KB      | 53053 KB     |
| iPhone 6 Plus                                          | 41800 KB      | 53082 KB     |
| iPhone 6                                               | 41773 KB      | 53053 KB     |
| iPhone 6S                                              | 41773 KB      | 53053 KB     |
| iPhone 6S Plus                                         | 41800 KB      | 53082 KB     |
| iPhone SE (1st generation)                             | 41773 KB      | 53053 KB     |
| iPhone 7                                               | 41773 KB      | 53053 KB     |
| iPhone 7 Plus                                          | 41800 KB      | 53082 KB     |
| iPhone 8                                               | 41773 KB      | 53053 KB     |
| iPhone 8 Plus                                          | 41800 KB      | 53082 KB     |
| iPhone X                                               | 41800 KB      | 53082 KB     |
| iPhone XS                                              | 41798 KB      | 53082 KB     |
| iPhone XS Max                                          | 41798 KB      | 53082 KB     |
| iPhone XR                                              | 41790 KB      | 53070 KB     |
| iPhone 11                                              | 22992 KB      | 52611 KB     |
| iPhone 11 Pro                                          | 23001 KB      | 52619 KB     |
| iPhone 11 Pro Max                                      | 23001 KB      | 52619 KB     |
| iPhone 12 mini                                         | 23001 KB      | 52619 KB     |
| iPhone 12                                              | 23001 KB      | 52619 KB     |
| iPhone 12 Pro                                          | 23001 KB      | 52619 KB     |
| iPhone 12 Pro Max                                      | 23001 KB      | 52619 KB     |

# 2.0.1 (Build 11) Staging

> Upgraded to Expo 41, Managed Workflow, non-EAS

| DEVICE TYPE                                            | DOWNLOAD SIZE | INSTALL SIZE |
| ------------------------------------------------------ | ------------- | ------------ |
| Universal                                              | 46634 KB      | 62782 KB     |
| iPad mini 3 Wi‑Fi + Cellular                           | 46669 KB      | 62475 KB     |
| iPad mini 4                                            | 46669 KB      | 62475 KB     |
| iPad mini 4 Wi‑Fi + Cellular                           | 46669 KB      | 62475 KB     |
| iPad Air 2                                             | 46669 KB      | 62475 KB     |
| iPad Air 2 Wi‑Fi + Cellular                            | 46669 KB      | 62475 KB     |
| iPad Pro (12.9‑inch)                                   | 46669 KB      | 62475 KB     |
| iPad (5th generation)                                  | 46669 KB      | 62475 KB     |
| iPad (5th generation) Wi‑Fi + Cellular                 | 46669 KB      | 62475 KB     |
| iPad Pro (12.9‑inch) (2nd generation)                  | 46669 KB      | 62475 KB     |
| iPad Pro (12.9‑inch) (2nd generation) Wi‑Fi + Cellular | 46669 KB      | 62475 KB     |
| iPad Pro (10.5‑inch)                                   | 46669 KB      | 62475 KB     |
| iPad Pro (10.5‑inch) Wi‑Fi + Cellular                  | 46669 KB      | 62475 KB     |
| iPad (6th generation)                                  | 46669 KB      | 62475 KB     |
| iPad (6th generation) Wi‑Fi + Cellular                 | 46669 KB      | 62475 KB     |
| iPad (7th generation) Wi‑Fi + Cellular                 | 24200 KB      | 54744 KB     |
| iPad Pro (11‑inch) (2nd generation) Wi‑Fi + Cellular   | 24200 KB      | 54744 KB     |
| iPad Pro (12.9‑inch) (4th generation)                  | 24200 KB      | 54744 KB     |
| iPad Pro (12.9‑inch) (4th generation) Wi‑Fi + Cellular | 24200 KB      | 54744 KB     |
| iPhone 5S                                              | 46669 KB      | 62475 KB     |
| iPhone 6 Plus                                          | 46696 KB      | 62504 KB     |
| iPhone 6                                               | 46669 KB      | 62475 KB     |
| iPhone 6S                                              | 46669 KB      | 62475 KB     |
| iPhone 6S Plus                                         | 46696 KB      | 62504 KB     |
| iPhone SE (1st generation)                             | 46669 KB      | 62475 KB     |
| iPhone 7                                               | 46669 KB      | 62475 KB     |
| iPhone 7 Plus                                          | 46696 KB      | 62504 KB     |
| iPhone 8                                               | 46669 KB      | 62475 KB     |
| iPhone 8 Plus                                          | 46696 KB      | 62504 KB     |
| iPhone X                                               | 46696 KB      | 62504 KB     |
| iPhone XS                                              | 46695 KB      | 62504 KB     |
| iPhone XS Max                                          | 46695 KB      | 62504 KB     |
| iPhone XR                                              | 46686 KB      | 62492 KB     |
| iPhone 11                                              | 24218 KB      | 54760 KB     |
| iPhone 11 Pro                                          | 24226 KB      | 54769 KB     |
| iPhone 11 Pro Max                                      | 24226 KB      | 54769 KB     |
| iPhone SE (2nd generation)                             | 24200 KB      | 54744 KB     |
| iPhone 12 mini                                         | 24226 KB      | 54769 KB     |
| iPhone 12                                              | 24226 KB      | 54769 KB     |
| iPhone 12 Pro                                          | 24226 KB      | 54769 KB     |
| iPhone 12 Pro Max                                      | 24226 KB      | 54769 KB     |
| iPod touch (6th generation)                            | 46669 KB      | 62475 KB     |
| iPod touch (7th generation)                            | 44333 KB      | 55244 KB     |

# 2.0.1 (Build 13) Staging

> Expo 41, Managed Workflow, EAS build

| DEVICE TYPE                                            | DOWNLOAD SIZE | INSTALL SIZE |
| ------------------------------------------------------ | ------------- | ------------ |
| Universal                                              | 3752 KB       | 6554 KB      |
| iPad mini 4                                            | 3572 KB       | 6337 KB      |
| iPad mini 4 Wi‑Fi + Cellular                           | 3572 KB       | 6337 KB      |
| iPad Air 2                                             | 3572 KB       | 6337 KB      |
| iPad Air 2 Wi‑Fi + Cellular                            | 3572 KB       | 6337 KB      |
| iPad (5th generation)                                  | 3572 KB       | 6337 KB      |
| iPad (5th generation) Wi‑Fi + Cellular                 | 3572 KB       | 6337 KB      |
| iPad Pro (12.9‑inch) (2nd generation)                  | 3572 KB       | 6332 KB      |
| iPad Pro (12.9‑inch) (2nd generation) Wi‑Fi + Cellular | 3572 KB       | 6332 KB      |
| iPad Pro (10.5‑inch)                                   | 3572 KB       | 6332 KB      |
| iPad Pro (10.5‑inch) Wi‑Fi + Cellular                  | 3572 KB       | 6332 KB      |
| iPad (6th generation)                                  | 3572 KB       | 6337 KB      |
| iPad (6th generation) Wi‑Fi + Cellular                 | 3572 KB       | 6337 KB      |
| iPad (7th generation) Wi‑Fi + Cellular                 | 1868 KB       | 6287 KB      |
| iPad Pro (11‑inch) (2nd generation) Wi‑Fi + Cellular   | 1867 KB       | 6291 KB      |
| iPad Pro (12.9‑inch) (4th generation)                  | 1867 KB       | 6291 KB      |
| iPad Pro (12.9‑inch) (4th generation) Wi‑Fi + Cellular | 1867 KB       | 6291 KB      |
| iPhone 5S                                              | 3551 KB       | 6312 KB      |
| iPhone 6 Plus                                          | 3569 KB       | 6332 KB      |
| iPhone 6                                               | 3551 KB       | 6312 KB      |
| iPhone 6S                                              | 3551 KB       | 6312 KB      |
| iPhone 6S Plus                                         | 3569 KB       | 6332 KB      |
| iPhone SE (1st generation)                             | 3551 KB       | 6312 KB      |
| iPhone 7                                               | 3551 KB       | 6312 KB      |
| iPhone 7 Plus                                          | 3569 KB       | 6332 KB      |
| iPhone 8                                               | 3551 KB       | 6312 KB      |
| iPhone 8 Plus                                          | 3569 KB       | 6332 KB      |
| iPhone X                                               | 3569 KB       | 6332 KB      |
| iPhone XS                                              | 3565 KB       | 6324 KB      |
| iPhone XS Max                                          | 3565 KB       | 6324 KB      |
| iPhone XR                                              | 3565 KB       | 6324 KB      |
| iPhone 11                                              | 1865 KB       | 6283 KB      |
| iPhone 11 Pro                                          | 1865 KB       | 6283 KB      |
| iPhone 11 Pro Max                                      | 1865 KB       | 6283 KB      |
| iPhone SE (2nd generation)                             | 1846 KB       | 6263 KB      |
| iPhone 12 mini                                         | 1865 KB       | 6283 KB      |
| iPhone 12                                              | 1865 KB       | 6283 KB      |
| iPhone 12 Pro                                          | 1865 KB       | 6283 KB      |
| iPhone 12 Pro Max                                      | 1865 KB       | 6283 KB      |
| iPod touch (6th generation)                            | 3551 KB       | 6312 KB      |
| iPod touch (7th generation)                            | 3546 KB       | 6304 KB      |

**Comment by brentvatne** [comment]
nice :)

**Comment by cheeaun** [comment]
# 2.0.2 (Build 24) Production

> Expo 43, Managed Workflow, EAS build



DEVICE TYPE | DOWNLOAD SIZE | INSTALL SIZE
-- | -- | --
Universal | 6.54 MB | 15.0 MB
iPad mini 4 | 6.31 MB | 14.8 MB
iPad mini 4 Wi‑Fi + Cellular | 6.31 MB | 14.8 MB
iPad Air 2 | 6.31 MB | 14.8 MB
iPad Air 2 Wi‑Fi + Cellular | 6.31 MB | 14.8 MB
iPad (5th generation) | 6.31 MB | 14.8 MB
iPad (5th generation) Wi‑Fi + Cellular | 6.31 MB | 14.8 MB
iPad Pro (12.9‑inch) (2nd generation) | 6.31 MB | 14.8 MB
iPad Pro (12.9‑inch) (2nd generation) Wi‑Fi + Cellular | 6.31 MB | 14.8 MB
iPad Pro (10.5‑inch) | 6.31 MB | 14.8 MB
iPad Pro (10.5‑inch) Wi‑Fi + Cellular | 6.31 MB | 14.8 MB
iPad (6th generation) | 6.31 MB | 14.8 MB
iPad (6th generation) Wi‑Fi + Cellular | 6.31 MB | 14.8 MB
iPad Pro (11‑inch) (2nd generation) Wi‑Fi + Cellular | 2.08 MB | 7.12 MB
iPad Pro (12.9‑inch) (4th generation) | 2.08 MB | 7.12 MB
iPad Pro (12.9‑inch) (4th generation) Wi‑Fi + Cellular | 2.08 MB | 7.12 MB
iPad (9th generation) | 2.08 MB | 7.12 MB
iPad (9th generation) Wi-Fi + Cellular | 2.08 MB | 7.12 MB
iPad Pro (12.9-inch) (5th generation) Wi-Fi + Cellular | 2.08 MB | 7.12 MB
iPad mini (6th generation) | 2.08 MB | 7.12 MB
iPad mini (6th generation) Wi-Fi + Cellular | 2.08 MB | 7.12 MB
iPhone 5S | 6.29 MB | 14.8 MB
iPhone 6 | 6.29 MB | 14.8 MB
iPhone 6S | 6.29 MB | 14.8 MB
iPhone 6S Plus | 6.31 MB | 14.8 MB
iPhone SE (1st generation) | 6.29 MB | 14.8 MB
iPhone 7 | 6.29 MB | 14.8 MB
iPhone 7 Plus | 6.31 MB | 14.8 MB
iPhone 8 | 6.29 MB | 14.8 MB
iPhone 8 Plus | 6.31 MB | 14.8 MB
iPhone X | 6.31 MB | 14.8 MB
iPhone XS | 6.31 MB | 14.8 MB
iPhone XS Max | 6.31 MB | 14.8 MB
iPhone XR | 6.31 MB | 14.8 MB
iPhone 11 | 2.08 MB | 7.11 MB
iPhone 11 Pro | 2.08 MB | 7.12 MB
iPhone 11 Pro Max | 2.08 MB | 7.12 MB
iPhone 12 mini | 2.08 MB | 7.12 MB
iPhone 12 | 2.08 MB | 7.12 MB
iPhone 12 Pro | 2.08 MB | 7.12 MB
iPhone 12 Pro Max | 2.08 MB | 7.12 MB
iPhone 13 Pro | 2.08 MB | 7.12 MB
iPhone 13 Pro Max | 2.08 MB | 7.12 MB
